### PR TITLE
Add Max extension method

### PR DESCRIPTION
**Pull request by dimitrov-d**

### CODE REVIEW

File: lib/src/extension/Max.dart [hunk 1]
```diff
@@ -0,0 +1,12 @@
+extension MaxFart<T> on Iterable<T> {
+  Iterable<T> Max(Iterable<T> iterable) {
+
+    var max = iterable == null || iterable.isEmpty ? null : iterable.first;
```
Comment (jkhmtx) [inline lib/src/extension/Max.dart]: Thank you for your contribution!

Just FYI, your implementation is sound, but there is simpler way build this ternary:

```dart
final isEmpty = iterable?.isEmpty ?? true;
var max =  isEmpty ? null : iterable.first;
```

I moved the `isEmpty` boolean to a separate line to more easily demonstrate the usage of the [null aware operator](https://medium.com/@thinkdigitalsoftware/null-aware-operators-in-dart-53ffb8ae80bb).

Please consider also creating a unit test for your submission!

**Review by jkhmtx — DISMISSED** [review]
Thanks for your submission!

**Comment by jkhmtx** [comment]
@dimitrov-d 

Please merge or rebase master into your branch so that we may proceed with this PR.

**Comment by dimitrov-d** [comment]
I've added some additional work based on your previous comments

**Comment by jkhmtx** [comment]
Hello @dimitrov-d , thanks for all your hard work on this PR!

I am very glad you added a unit test, because it has brought to light an issue that we appear to be having with your implementation. According to your test output, you cannot apply the `>` operator to an object of type `Object`. This means that your generic parameter `T` is not correctly constrained to types that can be compared with `>`.

Please update your implementation to use a type constraint as with the following example:

```dart
extension MaxFart<T extends num> on Interable<T> {
   // ...
}
```

**Review by jkhmtx — CHANGES_REQUESTED** [review]
Please get your tests passing (see earlier comment on how), and then this will be ready to go. You can use the command `pub run test` to test locally before committing and pushing your changes upstream.

**Comment by dimitrov-d** [comment]
> Please get your tests passing (see earlier comment on how), and then this will be ready to go. You can use the command `pub run test` to test locally before committing and pushing your changes upstream.

Ready to go

**Review by jkhmtx — APPROVED** [review]

**Comment by jkhmtx** [comment]
@dimitrov-d Very nice

**Pull request merged**